packages/clerk-js/src/ui/components/UserProfile/EmailForm.tsx

```diff
@@ -21,10 +22,9 @@ export const EmailForm = withCardStateProvider((props: EmailFormProps) => {
   const { user } = useUser();
   const environment = useEnvironment();
   const preferEmailLinks = emailLinksEnabledForInstance(environment);
-
   const [createEmailAddress] = useReverification(() => user?.createEmailAddress({ email: emailField.value }));
-
   const emailAddressRef = React.useRef<EmailAddressResource | undefined>(user?.emailAddresses.find(a => a.id === id));
+  const strategy = getVerificationStrategy(emailAddressRef.current, preferEmailLinks);
```

Comment:
TODO: We could map each component based on the strategy instead of performing multiple conditionals. 

Also should take a look at not using the ref, but actually using the email within user settings.